### ISSUE TITLE
JoinGame Exception

### ISSUE THREAD

**Issue by jtJava** [retrooper/packetevents#710]
**Describe the bug**
An exception pops up in the console, there's a wrapper being written/read wrong

**Software brand**
Please let us know what software you are using.
Velocity 3.3

**Plugins**
Please give us a list of your plugins. Sometimes we run into incompatibilities. Having an overview of the plugins on your server can really help identify the problem quicker.
Eg. ViaVersion, ProtocolLib, HolographicDisplays
You may also provide a screenshot.

**How To Reproduce**
No idea

**Expected behavior**
No error

**Screenshots**
```
[00:18:12 WARN] [com.github.retrooper.packetevents.PacketEventsAPI]: PacketEvents caught an unhandled exception while calling your listener.
java.lang.IndexOutOfBoundsException: readerIndex(14) + length(1) exceeds writerIndex(14): PooledUnsafeDirectByteBuf(ridx: 14, widx: 14, cap: 256)
        at io.netty.buffer.AbstractByteBuf.checkReadableBytes0(AbstractByteBuf.java:1442) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at io.netty.buffer.AbstractByteBuf.readByte(AbstractByteBuf.java:730) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at io.github.retrooper.packetevents.impl.netty.buffer.ByteBufOperatorImpl.readByte(ByteBufOperatorImpl.java:74) ~[?:?]
        at com.github.retrooper.packetevents.netty.buffer.ByteBufHelper.readByte(ByteBufHelper.java:67) ~[?:?]
        at com.github.retrooper.packetevents.wrapper.PacketWrapper.readByte(PacketWrapper.java:312) ~[?:?]
        at com.github.retrooper.packetevents.wrapper.PacketWrapper.readBoolean(PacketWrapper.java:324) ~[?:?]
        at com.github.retrooper.packetevents.wrapper.play.server.WrapperPlayServerJoinGame.read(WrapperPlayServerJoinGame.java:173) ~[?:?]
        at com.github.retrooper.packetevents.wrapper.PacketWrapper.readEvent(PacketWrapper.java:238) ~[?:?]
        at com.github.retrooper.packetevents.wrapper.PacketWrapper.<init>(PacketWrapper.java:143) ~[?:?]
        at com.github.retrooper.packetevents.wrapper.PacketWrapper.<init>(PacketWrapper.java:133) ~[?:?]
        at com.github.retrooper.packetevents.wrapper.play.server.WrapperPlayServerJoinGame.<init>(WrapperPlayServerJoinGame.java:60) ~[?:?]
        at com.github.retrooper.packetevents.manager.InternalPacketListener.onPacketSend(InternalPacketListener.java:102) ~[?:?]
        at com.github.retrooper.packetevents.event.PacketSendEvent.call(PacketSendEvent.java:51) ~[?:?]
        at com.github.retrooper.packetevents.event.EventManager.callEvent(EventManager.java:52) ~[?:?]
        at io.github.retrooper.packetevents.handlers.PacketEventsEncoder.read(PacketEventsEncoder.java:46) ~[?:?]
        at io.github.retrooper.packetevents.handlers.PacketEventsEncoder.encode(PacketEventsEncoder.java:68) ~[?:?]
        at io.github.retrooper.packetevents.handlers.PacketEventsEncoder.encode(PacketEventsEncoder.java:32) ~[?:?]
        at io.netty.handler.codec.MessageToByteEncoder.write(MessageToByteEncoder.java:107) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:881) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:863) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:968) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:856) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at io.netty.handler.codec.MessageToByteEncoder.write(MessageToByteEncoder.java:113) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:881) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:863) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:968) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:856) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at io.netty.channel.DefaultChannelPipeline.write(DefaultChannelPipeline.java:1015) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at io.netty.channel.AbstractChannel.write(AbstractChannel.java:301) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at com.velocitypowered.proxy.connection.MinecraftConnection.delayedWrite(MinecraftConnection.java:249) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at com.velocitypowered.proxy.connection.client.ClientPlaySessionHandler.handleBackendJoinGame(ClientPlaySessionHandler.java:523) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at com.velocitypowered.proxy.connection.backend.TransitionSessionHandler.lambda$handle$0(TransitionSessionHandler.java:133) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:787) ~[?:?]
        at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482) ~[?:?]
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:413) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[velocity.jar:3.3.0-SNAPSHOT (git-b9b11665-b343)]
        at java.lang.Thread.run(Thread.java:840) [?:?]
```

**Comment by jtJava** [comment]
It's possible the player is on 1.7.10 but not 100% sure

**Comment by retrooper** [comment]
Please find out what version this player is on, if this can be actively reproduced it'll help us.

**Comment by retrooper** [comment]
Probably fixed, please re-open if persists on 2.3.0
